modules/asm/versions.tf

```diff
@@ -23,6 +23,11 @@ terraform {
       source  = "hashicorp/kubernetes"
       version = "~> 2.0"
     }
+    google = {
+      source = "hashicorp/google"
+      # Avoid v4.49.0 for https://github.com/hashicorp/terraform-provider-google/issues/13507
+      version = ">= 4.47.0, < 4.49.0"
```

Comment:
Thanks for the PR @jonchen727 Can we do instead https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/7f67bcd8529528e89c544ad810c21f3c88f55080/modules/auth/versions.tf#L24

This will allow newer provider versions `> 4.50` to be picked up which has the fix https://github.com/GoogleCloudPlatform/magic-modules/pull/7162

Reply:
I have added the changes as well as added the version constraint to fleet-membership which also uses `data "google_container_cluster"`